### ISSUE TITLE
A loading state for plugins

### ISSUE THREAD

**Issue by jadjoubran** [jadjoubran/webdash#77]
using [paper-progress](https://www.webcomponents.org/element/PolymerElements/paper-progress)

that shows a loader at the top of the plugin
and can be enabled/disabled from the plugins

feedback?

**Comment by adithyaChander** [comment]
Hey, Is this issue open? If it is, could I please give it try ? This looks really interesting.

**Comment by jadjoubran** [comment]
Hey @adithyaChander 
Yes indeed this is still open 😄 
Feel free to submit a PR and also feel free to ask me for help throughout if you've got any questions, I'd love to help
Cheers!

**Comment by adithyaChander** [comment]
Hey @jadjoubran . Thanks a lot ! :) 
I will work on this and submit a PR. This is really exciting.  :)

**Comment by jadjoubran** [comment]
Thank you! Don't hesitate to ask for help as this one might be a bit tricky
Cheers!

**Comment by adithyaChander** [comment]
Yeah, sure. Any input is much appreciated. I am going through the code to understand how the application works. If you think I should start from a specific place, kindly let me know. Thank you once again.

**Comment by jadjoubran** [comment]
I did a quick research and we're most likely going to have to do something similar to the below:

`<plugin-wrapper>` and the `plugin` that's being loaded in it need to communicate. Either via events or by 1 or 2 way data binding to let the plugin signal that it's in a `loading` state
And whenever the `<plugin-wrapper>` is in a loading state, then we show the loader

I'm still not sure exactly how, if you get stuck, I can do further research

plugin-wrapper is in `src/plugin-wrapper.html`, I think this is where most of the changes will be
and you might need to play with a sample plugin (to trigger the loading state), you can clone any plugin I've used or setup a [new plugin from scratch](https://github.com/jadjoubran/webdash/wiki/Create-a-plugin)
